File: client/web/sudoo/lib/domain/model/discovery/category.dart


class Category {
  final String categoryId;
  String name;
  String image;
  bool enable = true;
  final int? countProduct;

  Category(
    this.categoryId,
    this.name,
    this.image,
      this.enable, {
    this.countProduct,
  });

  factory Category.empty() {
    return Category("", "", "", true);
  }
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/discovery/Category.kt
package com.sudo248.sudoo.domain.entity.discovery

import com.sudo248.base_android.base.ItemDiff


/**
 * **Created by**
 *
 * @author *Sudo248*
 * @since 09:43 - 12/03/2023
 */
data class Category(
    val categoryId: String,
    val name: String,
    val image: String,
    var products: MutableList<ProductInfo>? = null,
) : java.io.Serializable, ItemDiff {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is Category && this == other
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is Category && categoryId == other.categoryId
    }

}